Spigot-API-Patches/0127-Expand-ArmorStand-API.patch

```diff
@@ -0,0 +1,109 @@
+From 580e567a9fe8dc228bf8706435cf98f7adc16389 Mon Sep 17 00:00:00 2001
+From: willies952002 <admin@domnian.com>
+Date: Thu, 26 Jul 2018 02:22:44 -0400
+Subject: [PATCH] Expand ArmorStand API
+
+Add the following:
+- Add proper methods for getting and setting items in both hands. Deprecates old methods
+- Enable/Disable slot interactions
+
+diff --git a/src/main/java/org/bukkit/entity/ArmorStand.java b/src/main/java/org/bukkit/entity/ArmorStand.java
+index 099da6ce..d204c535 100644
+--- a/src/main/java/org/bukkit/entity/ArmorStand.java
++++ b/src/main/java/org/bukkit/entity/ArmorStand.java
+@@ -1,8 +1,11 @@
+ package org.bukkit.entity;
+ 
++import org.bukkit.inventory.EquipmentSlot; // Paper
+ import org.bukkit.inventory.ItemStack;
+ import org.bukkit.util.EulerAngle;
+ 
++import java.util.List; // Paper
++
+ public interface ArmorStand extends LivingEntity {
+ 
+     /**
+@@ -10,6 +13,10 @@ public interface ArmorStand extends LivingEntity {
+      * currently holding
+      *
+      * @return the held item
++     // Paper start - Deprecate in favor of setItemInMainHand
++     * @deprecated use {@link ArmorStand#getItemInMainHand()} instead
++     * @see ArmorStand#getItemInMainHand()
++     // Paper end
+      */
+     ItemStack getItemInHand();
+ 
+@@ -18,6 +25,10 @@ public interface ArmorStand extends LivingEntity {
+      * holding
+      *
+      * @param item the item to hold
++     // Paper start - Deprecate in favor of setItemInMainHand
++     * @deprecated use {@link ArmorStand#setItemInMainHand(ItemStack)} instead
++     * @see ArmorStand#setItemInMainHand(ItemStack)
++     // Paper end
+      */
+     void setItemInHand(ItemStack item);
+ 
+@@ -275,5 +286,58 @@ public interface ArmorStand extends LivingEntity {
+      * @param move {@code true} if this armour stand can move, {@code false} otherwise
+      */
+     void setCanMove(boolean move);
++
++    /**
++     * Returns the item the armor stand is
++     * currently holding in it's main hand
++     *
++     * @return the held item
++     */
++    ItemStack getItemInMainHand();
++
++    /**
++     * Sets the item the armor stand is currently
++     * holding in it's main hand
++     *
++     * @param item the item to hold
++     */
++    void setItemInMainHand(ItemStack item);
```

Comment:
I'd like to see a method like https://github.com/PaperMC/Paper/blob/master/Spigot-Server-Patches/0091-Access-items-by-EquipmentSlot.patch#L39-L47:
```java
ItemStack setItemInHand(final org.bukkit.inventory.EquipmentSlot slot, final ItemStack item);
```